### PR TITLE
Export coverage report to codecov

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Comment by bogdandrutu** [comment]
The result is unexpected, and is on the bad side :). I think a reasonable repository should have 80+ coverage, I do understand that there are errors hard to test but still 80% is reasonable as a goal

File: .circleci/config.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ build-template: &build-template
         name: "Precommit and Coverage Report"
         command: |
           make ci
+          bash <(curl -s https://codecov.io/bash)
```
Comment (MrAlias) [inline .circleci/config.yml:28]: I would much prefer to use a versioned build step to piping a downloaded bash script: https://circleci.com/docs/2.0/code-coverage/#codecov

Comment (evantorrie) [reply .circleci/config.yml]: @MrAlias I changed this to use the codecov orb.  Take another look.

**Comment by Aneurysm9** [comment]
`go tool cover` appears to be hung up on changes that have added packages to sub-modules since the last release.  Does this need to generate separate coverage reports for each module?

**Comment by MrAlias** [comment]
> `go tool cover` appears to be hung up on changes that have added packages to sub-modules since the last release. Does this need to generate separate coverage reports for each module?

Generation of each sub-module sounds correct to me.

**Comment by evantorrie** [comment]
Yes, I'll look at this again.

**Comment by evantorrie** [comment]
@MrAlias Yes, ready to merge now.

**Pull request merged**